### ISSUE TITLE
Unable find the video download link

### ISSUE THREAD

**Issue by ADEELAHMAD256** [thitlwincoder/direct_link#23]
HI bro i am not finding the correct video download link 
like         print("Link.....................${e.link}"); code take form readme file. 
Thumbnail URL working correctly fine.

**Comment by thitlwincoder** [comment]
Hi @ADEELAHMAD256. What is the URL used and what platform?

**Comment by qimar** [comment]
Hi @thitlwincoder, 

Great package, but it is not consistant. 

Im also facing issue package parsing html on low level code is not fetching correct html elements from  where video meta data information can be fetched. See the screen shot of the file social.dart 

Your package is not parsing this part of the html:
![image](https://github.com/user-attachments/assets/e36e0b3e-d307-4886-b23d-fbb02d6056e7)


I tried this facebook link:

https://www.facebook.com/share/v/qGQfwEUtQasJKyPC/?mibextid=jmPrMh
![Screenshot_1](https://github.com/user-attachments/assets/efcddcaa-c687-42ad-a24b-02ce4a4d228c)

**Comment by thitlwincoder** [comment]
@ADEELAHMAD256 I tried with your given URL. This is a result.
![Screenshot_1725191731](https://github.com/user-attachments/assets/e154ad1b-373a-4f36-a9dd-195fe945b9df)

**Comment by qimar** [comment]
Yes thats the video but the problem is that i am not recieving the meta data in my app, even when the website has generated the data.

**Comment by marcelov2** [comment]
I tried but it didn't work for me or I did something wrong

![Image](https://github.com/user-attachments/assets/141464ab-935c-4f11-b72a-b2d19a48ea69)

link = https://vkvideo.ru/video669319989_456239048
link = https://youtu.be/dpppZ9ySJSY?si=woOjnJDRwNT0qYng

**Comment by clasher42** [comment]
The link that the package is extracting is an image file, not an .mp4 file. Please try to fix this issue.

here is the URL that the package is fetching:
[{url: https://media.sf-converter.com/get?__sig=EGN_0hKzJG4kfWkjLDhiYw&__expires=1741782174&uri=https%3A%2F%2Finstagram.fruh2-2.fna.fbcdn.net%2Fv%2Ft51.2885-15%2F483859432_17874183117291912_5164435039448407688_n.jpg%3Fstp%3Ddst-jpg_e15_tt6%26_nc_ht%3Dinstagram.fruh2-2.fna.fbcdn.net%26_nc_cat%3D100%26_nc_oc%3DQ6cZ2AESH52nKkvzFqMDclTmgJr-TYtBljijtDoe3GrRMVhRT_Tx2er7tEYjxHMUGYajoPEx7g49SGhkiLmw3wtIjRn6%26_nc_ohc%3DiX9Dr3Q-FFQQ7kNvgFGiFw7%26_nc_gid%3D8224b66e2fa04ab4a4ce638c199374c4%26edm%3DANTKIIoBAAAA%26ccb%3D7-5%26oh%3D00_AYEPBGI_uC1IurFDA_wndlGYMDQPCiZUAB7RvAvzQYMHxA%26oe%3D67D70380%26_nc_sid%3Dd885a2&filename=483859432_17874183117291912_5164435039448407688_n.jpg, initiatorType: img, startTime: 10948.5, duration: 1820}]

**Comment by thitlwincoder** [comment]
@clasher42, what is your test url?

**Comment by clasher42** [comment]
@thitlwincoder  i have tested on Instagram reel, i have tried two URLs:
https://www.instagram.com/share/reel/BATokH5llc
https://www.instagram.com/viralbhayani/reel/C99I7bONc0R/

**Comment by thitlwincoder** [comment]
@clasher42 Try again with the latest version. Now I updated it.
